ArcGISRuntimeSDKQt_CppSamples/Maps/ApplyScheduledMapUpdates/ApplyScheduledMapUpdates.cpp

```diff
@@ -121,39 +135,47 @@ MapQuickView* ApplyScheduledMapUpdates::mapView() const
   return m_mapView;
 }
 
-// Set the view (created in QML)
-void ApplyScheduledMapUpdates::setMapView(MapQuickView* mapView)
+void ApplyScheduledMapUpdates::setMapToMapView()
 {
-  if (!mapView || mapView == m_mapView)
+  if (!m_map || !m_mapView)
     return;
 
-  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+}
 
-  // create MMPK
-  qDebug() << m_TempDir.path() + "/" + sampleMmpk;
-  m_mobileMapPackage = new MobileMapPackage(m_TempDir.path() + "/" + sampleMmpk, this);
+void ApplyScheduledMapUpdates::onMmpkDoneLoading(Error e)
+{
+  // check if successful
+  if (!e.isEmpty())
+    return;
 
-  // load MMPK
-  connect(m_mobileMapPackage, &MobileMapPackage::doneLoading, this, [this](Error e)
-  {
-    if (!e.isEmpty())
-      return;
+  // make sure there are valid maps
+  if (m_mobileMapPackage->maps().isEmpty())
+    return;
 
-    // set the map on the map view
-    m_map = m_mobileMapPackage->maps().at(0);
-    m_mapView->setMap(m_map);
+  // set the map on the map view
+  m_map = m_mobileMapPackage->maps().at(0);
+  setMapToMapView();
 
-    // setup sync task
-    m_offlineSyncTask = new OfflineMapSyncTask(m_map, this);
+  // setup sync task
+  m_offlineSyncTask = new OfflineMapSyncTask(m_map, this);
```

Comment:
I guess this would leak the 2nd time around (e.g. if the mmpk gets recreated). You could check if it's `nullptr` before creating?

Reply:
yeah good catch